qrenderdoc/Windows/DebugMessageView.cpp

```diff
@@ -369,9 +374,21 @@ void DebugMessageView::messages_contextMenu(const QPoint &pos)
     m_ToggleMessageType->setText(tr("%1 Message Type").arg(hidden ? show : hide));
 
     m_ContextMessage = msg;
-
-    RDDialog::show(m_ContextMenu, ui->messages->viewport()->mapToGlobal(pos));
   }
+  else
+  {
+    m_ToggleSource->setVisible(false);
+    m_ToggleSeverity->setVisible(false);
+    m_ToggleCategory->setVisible(false);
+    m_ToggleMessageType->setVisible(false);
+  }
+
+  QString showHidden = tr("Show hidden rows");
+  QString hideHidden = tr("Stop showing hidden rows");
+
+  m_ShowHidden->setText(m_FilterModel->showHidden ? hideHidden : showHidden);
```

Comment:
I think this change might be a reaction to a different bug. The `Show hidden rows` item is supposed to have a check mark which shows up and disappears depending on if it's enabled - see `setCheckable` and `setChecked`. However in the custom style code though it looks like I don't yet handle menu items with checkmarks correctly so it's never displayed - if you toggle to the native theme you should see what I mean.

I'll fix this either way since it's actually used elsewhere in a couple of places. It's something about the phrasing of `Stop showing hidden rows` that made me pause - it's not exactly a double negative it does take a bit of unravelling. I think it would be better to keep a simple statement `Show hidden rows` and then show the status of it with the check, once that's fixed.

Reply:
I figured this was the case, but I'm unfamilar with QT and I don't have the source for it locally so I couldn't investigate further. A check mark would definitely be better than switching the text.

Reply:
No problem. I've pushed my changes to display the checkbox so you should be able to rebase onto latest and pick it up.